src/lib/context.rs

```diff
@@ -7,15 +7,18 @@ use std::path::PathBuf;
 #[derive(Clone, Debug)]
 pub struct Context {
     pub path: Option<PathBuf>,
-    pub line_number: usize,
-    // pub row_number: Option(usize),
+    pub line_number: u64,
```

Comment:
Would changing this to `u64` prevent compiling on 32-bit platform?

Reply:
Actually most 32bit hardware can deal with 64bit integers in hardware, and rust seems to implement them in software for those that do not, you can test this by building this branch with these:
```bash
$ rustup target add i686-unknown-linux-gnu
$ cargo build --target i686-unknown-linux-gnu
```

Reply:
That's a good point. The reason I chose `usize` is because `enumerate` returns `usize`. Why would we switch to `u64`?

Reply:
mainly because usize is pointer size and platform dependent, setting it to u64 or u32 is more explicit in this case to the intent of this code.

Reply:
Fair enough; I think the reason I normally would prefer `usize` is so it matches what `enumerate` returns for the index. Granted, it's not likely to be a practical issue because if/when 128-bit platforms start becoming more popular it's not like we'll have to deal with make files that have more than `2^64` lines, so I'm fine with this change.

Reply:
Should we mark this point as resolved then?